crates/router/src/connector/bluesnap/transformers.rs

```diff
@@ -0,0 +1,349 @@
+    AuthReversal,
+    Capture,
+    Refund,
+}
+
+#[derive(Debug, Clone, Default, Serialize, Deserialize, PartialEq)]
+pub enum BluesnapProcessingStatus {
+    #[serde(rename = "success")]
+    SuccessLowercase,
+    #[serde(rename = "SUCCESS")]
+    SuccessUppercase,
+    #[default]
+    #[serde(rename = "pending")]
+    PendingLowercase,
+    #[serde(rename = "PENDING")]
+    PendingUppercase,
+    #[serde(rename = "fail")]
+    FailLowercase,
+    #[serde(rename = "FAIL")]
+    FailUppercase,
+    #[serde(rename = "pending_merchant_review")]
+    PendingMerchantReviewLowercase,
+    #[serde(rename = "PENDING_MERCHANT_REVIEW")]
+    PendingMerchantReviewUppercase,
+}
+
+impl TryFrom<transformers::Foreign<(BluesnapTxnType, BluesnapProcessingStatus)>>
+    for transformers::Foreign<enums::AttemptStatus>
+{
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        item: transformers::Foreign<(BluesnapTxnType, BluesnapProcessingStatus)>,
+    ) -> Result<Self, Self::Error> {
+        let item = item.0;
+        let (item_txn_status, item_processing_status) = item;
+        Ok(match item_processing_status {
+            BluesnapProcessingStatus::SuccessLowercase
+            | BluesnapProcessingStatus::SuccessUppercase => match item_txn_status {
+                BluesnapTxnType::AuthOnly => enums::AttemptStatus::Authorized,
+                BluesnapTxnType::AuthReversal => enums::AttemptStatus::Voided,
+                BluesnapTxnType::AuthCapture | BluesnapTxnType::Capture => {
+                    enums::AttemptStatus::Charged
+                }
+                BluesnapTxnType::Refund => enums::AttemptStatus::Charged,
+            },
+            BluesnapProcessingStatus::PendingLowercase
+            | BluesnapProcessingStatus::PendingUppercase
+            | BluesnapProcessingStatus::PendingMerchantReviewLowercase
+            | BluesnapProcessingStatus::PendingMerchantReviewUppercase => {
+                enums::AttemptStatus::Pending
+            }
+            BluesnapProcessingStatus::FailLowercase | BluesnapProcessingStatus::FailUppercase => {
+                enums::AttemptStatus::Failure
+            }
+        }
+        .into())
+    }
+}
+
+impl TryFrom<transformers::Foreign<(BluesnapTxnType, BluesnapProcessingStatus)>>
+    for transformers::Foreign<enums::RefundStatus>
+{
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        item: transformers::Foreign<(BluesnapTxnType, BluesnapProcessingStatus)>,
+    ) -> Result<Self, Self::Error> {
+        let item = item.0;
+        let (item_txn_status, item_processing_status) = item;
+        Ok(match item_processing_status {
+            BluesnapProcessingStatus::SuccessLowercase
+            | BluesnapProcessingStatus::SuccessUppercase => match item_txn_status {
+                BluesnapTxnType::Refund => enums::RefundStatus::Success,
+                _ => Err(errors::ConnectorError::ResponseHandlingFailed)
```

Comment:
why is this match necessary? since the transaction type will always be refund right?